File: php/features.php
<?php
declare(strict_types=1);

// Imgur SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class ImgurFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new ImgurBaseFeature();
            case "test":
                return new ImgurTestFeature();
            default:
                return new ImgurBaseFeature();
        }
    }
}

File: py/imgur_sdk/features.py
# Imgur SDK feature factory

from imgur_sdk.feature.base_feature import ImgurBaseFeature
from imgur_sdk.feature.test_feature import ImgurTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ImgurBaseFeature(),
        "test": lambda: ImgurTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
